debian/rules

```diff
@@ -20,6 +20,7 @@ extraopts += -DCMAKE_INSTALL_LIBDIR=/usr/lib
 extraopts += -DCMAKE_INSTALL_LIBEXECDIR=/usr/lib
 extraopts += -DCMAKE_INSTALL_SYSCONFDIR=/etc
 extraopts += -DCMAKE_INSTALL_SYSTEMD_SERVICEDIR=/lib/systemd/system
+extraopts += -DCMAKE_BUILD_TYPE=Release 
```

Comment:
i'd prefer not hardwiring this setting to Release, as it will apply to all building hosts which have enough memory for `-g` builds . probably you could define `CEPH_EXTRA_CMAKE_ARGS` instead?

Reply:
The problem is, that not a single 32-Bit host hast "enough" memory to compile it with GCC. With clang, the 3G-Limit is not exceeded, with GCC I have not found a version supporting C++17 that is not kicking this 32-Bit-Systems limit.

Reply:
@the78mole i am not against the build w/o `-g`, and i think i am aware of the issue you are facing. because we always need to generate the debug symbols for debugging as long as possible. what i suggest is to define the `CEPH_EXTRA_CMAKE_ARGS` environmental variable, and it will be consumed by `debian/rules` or whatever tools you use for building the deb packages. like
```sh
CEPH_EXTRA_CMAKE_ARGS="-DCMAKE_BUILD_TYPE=Release" debian/rules binary
# or 
CEPH_EXTRA_CMAKE_ARGS="-DCMAKE_BUILD_TYPE=Release" dpkg-buildpackage -b
```

could you consider this as an option?